tests/functional/object/mcg/test_endpoint_network_disruption.py

```
@@ -0,0 +1,384 @@
+        operation_thread.start()
+        time.sleep(2)
+
+        logger.info(
+            "Applying NetworkPolicy to block external egress "
+            "from noobaa-endpoint pods only"
+        )
+        network_policy_data = templating.load_yaml(
+            constants.TEMPLATE_BLOCK_NB_EGRESS_NETWORK_POLICY
+        )
+        network_policy_data["metadata"]["namespace"] = namespace
+        temp_yaml = tempfile.NamedTemporaryFile(
```

Ruff (0.15.14)

[warning] Use a context manager for opening files

(SIM115)